docs/architecture/0e-plans.md

```diff
@@ -590,3 +590,24 @@ https://github.com/Mbed-TLS/mbedtls/issues/8452
 
 ### Shrunk-down `psa_util.h`
 
+Only some of the interface elements in `mbedtls/psa_util.h` remain useful.
+
+| Name | Uses | Fate |
+| ---- | ---- | ---- |
+| `mbedtls_psa_get_random` | crypto internal, `ssl_test_lib.c` | Deprecate |
+| `MBEDTLS_PSA_RANDOM_STATE` | same as `mbedtls_psa_get_random` | Deprecate |
+| `mbedtls_ecc_group_to_psa` | crypto internal | Internalize |
+| `mbedtls_ecc_group_from_psa` | crypto internal | Internalize |
+| `mbedtls_md_psa_alg_from_type` | crypto internal, SSL library | Privatize, deprecate or remove? |
+| `mbedtls_md_type_from_psa_alg` | crypto internal, SSL library | Privatize, deprecate or remove? |
+| `mbedtls_ecdsa_raw_to_der` | crypto internal | Keep public |
+| `mbedtls_ecdsa_der_to_raw` | crypto internal | Keep public |
+
+The RNG wrapper function `mbedtls_psa_get_random` (and the associated constant `MBEDTLS_PSA_RANDOM_STATE`) has a negligible cost for us and can help users transition their code. Keep it, but mark it as deprecated, possibly inlined, and change it (code and documentation) to just return the return value of `psa_get_random`.
```

Comment:
> can help users transition their code

What do you have in mind here? My thinking is this function's primary purpose was it could be used as the `f_rng` argument of functions that have such a parameter. But we'll be removing all such parameters, so once that's done how will this function still be helpful to users?

Reply:
Anything that happens in X.509 or TLS as a result of API changes in crypto, is also something that's likely to happen in code that implements other protocols. Maintainers of third-party code bases that implement other protocols may want to start using TF-PSA-Crypto 1.0 before they have fully converted their code. They may even want to support both Mbed TLS 3.6 and TF-PSA-Crypto in the same code branch, if they deploy the same application to devices which are not all upgraded to the latest crypto at the same time. So they may want to pass `f_rng` arguments around in their own code, even if they end up being unused when building against TF-PSA-Crypto.